client/src/pages/Home/Home.jsx

```diff
@@ -1,8 +1,18 @@
 import { Box, makeStyles } from '@material-ui/core';
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useContext } from 'react';
+import { getFriends, getPolls } from '../../api/api';
 import { useQuery } from 'react-query';
-import { getPolls } from '../../api';
 import { FriendList, Friends, Polls } from '../../components';
+import { getFriendLists, getFriendInfo } from '../../api/api';
+import { AiOutlineConsoleSql } from 'react-icons/ai';
+import { GlobalContext } from '../../utils';
+import io from 'socket.io-client';
+
+const socket = io('/', {
+  query: {
+    token: localStorage.getItem('io'),
+  },
+});
```

Comment:
Do this setup in a different file that exports `socket`. Then import here and put it in a use effect that opens it when the component mounts, and disconnects it on unmount. Don't pass a token, let's use cookies instead